docs: re-add `AsyncAPI Documents` and change code snippet

animeshk923 opened the pull request:
## Description
- Re-added the `AsyncAPI Documents` in the next-major-spec branch.
- Changed the code snippet to match v3.

**Related issue(s)**
fixes #1858

animeshk923 committed:
re-add `AsyncAPI Documents`

changes:

- Re-added the `AsyncAPI Documents` in the next-major-spec branch.
- Changed the code snippet to match v3.

animeshk923 committed:
fixed code format

derberg commented on pages/docs/tutorials/getting-started/asyncapi-documents.md:
```diff
@@ -0,0 +1,48 @@
+      $ref: '#/channels/userSignup'`}
+</CodeBlock>
+
+The AsyncAPI document is a machine-readable definition of your Event-Driven API. This document can be used afterward to generate documentation and code, validate the messages your application receives, and even apply API management policies to your messages before they arrive to your broker.
```
```suggestion
The AsyncAPI document is a machine-readable definition of your Event-Driven API. This document can be used afterward to generate documentation and code, validate the messages that `Example application` sends, and even apply API management policies to your messages before they arrive to the broker.
```

animeshk923 replied:
resolved with 24fdfd8

derberg reviewed (commented):
@AnimeshKumar923 perfectly migrated AsyncAPI document. There is just one docu suggestion that you have to apply - just to make sure it is not about user application perspective but the perspective of the example app that this document describes

animeshk923 committed:
Update pages/docs/tutorials/getting-started/asyncapi-documents.md

suggestion applied from: https://github.com/asyncapi/website/pull/2274#discussion_r1386884048

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

animeshk923 commented:
> @AnimeshKumar923 perfectly migrated AsyncAPI document.

Thank you! :pray: 

> There is just one docu suggestion that you have to apply - just to make sure it is not about user application perspective but the perspective of the example app that this document describes

applied suggestion with 24fdfd8

derberg commented on pages/docs/tutorials/getting-started/asyncapi-documents.md:
```diff
@@ -0,0 +1,48 @@
+    action: 'send'
+    channel:
+      $ref: '#/channels/userSignup'`}
+</CodeBlock>
```
```suggestion
<CodeBlock>
{`asyncapi: 3.0.0
info:
  title: Example application
  version: '0.1.0'
channels:
  userSignup:
    address: 'user/signedup'
    messages:
      userSignedupMessage: 
        description: A message describing that a user just signed up.
        payload:
          type: object
          additionalProperties: false
          properties:
            fullName:
              type: string
            email:
              type: string
              format: email
            age:
              type: integer
              minimum: 18
operations:
  publishUserSignedup:
    action: 'send'
    channel:
      $ref: '#/channels/userSignup'`}
</CodeBlock>
```

small change, build fails as CodeBlock do not like empty spaces

animeshk923 replied:
> small change, build fails as CodeBlock do not like empty spaces

hmm...new thing to learn :thinking: 

resolved with a03a7bb

animeshk923 committed:
Update pages/docs/tutorials/getting-started/asyncapi-documents.md

applied suggestion from: https://github.com/asyncapi/website/pull/2274#discussion_r1387722670

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

quetzalliwrites commented on pages/docs/tutorials/getting-started/asyncapi-documents.md:
```diff
@@ -0,0 +1,46 @@
+
+An AsyncAPI document is a file that defines and annotates the different components of **a specific Event-Driven API**.
+
+The format of the file must be JSON or YAML; however, only the subset of YAML that matches the JSON capabilities is allowed.
```
```suggestion
The file format must be JSON or YAML; however, only the subset of YAML that matches the JSON capabilities is allowed.
```

quetzalliwrites commented on pages/docs/tutorials/getting-started/asyncapi-documents.md:
```diff
@@ -0,0 +1,46 @@
+      $ref: '#/channels/userSignup'`}
+</CodeBlock>
+
+The AsyncAPI document is a machine-readable definition of your Event-Driven API. This document can be used afterward to generate documentation and code, validate the messages that `Example application` sends, and even apply API management policies to your messages before they arrive to the broker.
```
```suggestion
The AsyncAPI document is a machine-readable definition of your Event-Driven API. That document can be used afterward to generate documentation and code, validate the messages that `Example application` sends, and even apply API management policies to your messages before they arrive at the broker.
```

quetzalliwrites commented on pages/docs/tutorials/getting-started/asyncapi-documents.md:
```diff
@@ -0,0 +1,46 @@
+
+The AsyncAPI document is a machine-readable definition of your Event-Driven API. This document can be used afterward to generate documentation and code, validate the messages that `Example application` sends, and even apply API management policies to your messages before they arrive to the broker.
+
+Your API documentation is now machine-readable –easily parseable by code— so the myriad of useful applications is endless.
```
```suggestion
Your API documentation is now machine-readable (easily parseable by code) so the myriad of useful applications is endless.
```

Commit:
tw editorial fixes

quetzalliwrites commented:
/rtm
